flex_container_orchestrator/domain/aggregator_flexpart.py

```diff
@@ -0,0 +1,340 @@
+    """
+    processed_items = set()
+    try:
+        cursor = conn.cursor()
+        for frt in frt_s:
+            cursor.execute(
+                """
+                SELECT processed, step FROM uploaded
+                WHERE forecast_ref_time = ?
+            """,
+                (frt,),
+            )
+            items_f = cursor.fetchall()
+            for item in items_f:
+                if item[0]:  # processed == True
+                    frt_str = (
+                        frt.strftime("%Y%m%d%H%M")
+                        if isinstance(frt, datetime.datetime)
+                        else str(frt)
+                    )
+                    processed_items.add(frt_str + f"{int(item[1]):02}")
+    except sqlite3.Error as e:
+        logging.error("SQLite query error while fetching processed items: %s", e)
+        sys.exit(1)
+    return processed_items
+
+
+def define_config(st: datetime.datetime, et: datetime.datetime) -> dict:
+    """
+    Define configuration for Flexpart.
+
+    Args:
+        st (datetime.datetime): Start time.
+        et (datetime.datetime): End time.
+
+    Returns:
+        dict: Configuration dictionary for Flexpart.
+    """
+    logging.info("Start and end time to configure Flexpart: %s and %s ", st, et)
+
+    configuration = {
+        "IBDATE": st.strftime("%Y%m%d"),
+        "IBTIME": st.strftime("%H"),
+        "IEDATE": et.strftime("%Y%m%d"),
+        "IETIME": et.strftime("%H"),
+    }
+
+    logging.info("Configuration to run Flexpart: %s", json.dumps(configuration))
+
+    return configuration
+
+
+# mypy: ignore-errors
+def get_time_settings(config) -> dict:
+    """
+    Retrieve time settings from config.
+
+    Args:
+        config: Configuration object.
+
+    Returns:
+        dict: Dictionary with time settings.
+    """
+    return {
+        "tincr": config.main.time_settings.tincr,
+        "tdelta": config.main.time_settings.tdelta,
+        "tfreq_f": config.main.time_settings.tfreq_f,
+        "tfreq": config.main.time_settings.tfreq,
+    }
+
+
+def parse_forecast_datetime(date_str: str, time_str: str) -> datetime.datetime:
+    """
+    Parse forecast date and time strings into a datetime object.
+
+    Args:
+        date_str (str): Date in YYYYMMDD format.
+        time_str (str): Time in HH format.
+
+    Returns:
+        datetime.datetime: Parsed datetime object.
+    """
+    return datetime.datetime.strptime(f"{date_str}{int(time_str):02d}00", "%Y%m%d%H%M")
+
+
+def generate_forecast_times(
+    list_start_times: List[datetime.datetime], time_settings: dict
+) -> Tuple[List[List[str]], List[List[datetime.datetime]], Set[str]]:
+    """
+    Generate forecast times for Flexpart runs.
+
+    Args:
+        list_start_times (list of datetime.datetime): List of Flexpart run start times.
+        time_settings (dict): Time settings dictionary.
+
+    Returns:
+        tuple: Tuple containing lists of forecast times, lead times, and all steps.
```

Comment:
We should describe here what is the difference between lead times and all steps, to me it is not immediately clear